tabpfn_client/client.py

```diff
@@ -107,20 +186,33 @@ def upload_train_set(self, X, y) -> str:
             The unique ID of the train set in the server.
 
         """
-
-        X = common_utils.serialize_to_csv_formatted_bytes(X)
-        y = common_utils.serialize_to_csv_formatted_bytes(y)
+        # Save until prediction for retrying train set upload for the case that anything went wrong with cache.
+        self._cache_manager.temporary_save_train_set(X, y)
+        X_serialized = common_utils.serialize_to_csv_formatted_bytes(X)
+        y_serialized = common_utils.serialize_to_csv_formatted_bytes(y)
+
+        # Get hash for dataset. Include access token for the case that one user uses different accounts.
+        dataset_hash = str(
```

Comment:
I think the hashing could very well go inside the `DatasetUIDCache`, the function should just accept the X,y and extra strings to incorporate (e.g. acccess_token or train_uid)

Reply:
I also thought about this, I was a bit hesitant to do that because we hash very different kind of stuff in fit and predict. But I agree with you. I moved the hashing to the DatesetUIDCacheManager and simply combine all arguments for the hash. Still not 100% sure how intuitive my new implementation is

Reply:
👍